application/heartbeater.go

```diff
@@ -10,13 +11,16 @@ import (
 // HeartbeatService sends a heartbeat metric every 5 minutes
 type HeartbeatService interface {
 	Close()
+	AddCustomTags(tags map[string]string)
 }
 
 type heartbeater struct {
 	sender      senders.Sender
 	application Tags
 	source      string
 	components  []string
+	customTags  []map[string]string
+	mux         sync.Mutex
```

Comment:
move this above the customTags. It's idiomatic in Go to define the mutex above the fields it provides mutually exclusive access to.